File: das_backend/src/main/java/ch/sbb/backend/admin/domain/repository/ServicePointRepository.java
package ch.sbb.backend.admin.domain.repository;

import ch.sbb.backend.admin.domain.model.ServicePoint;
import java.util.List;
import java.util.Optional;

public interface ServicePointRepository {

    Optional<ServicePoint> findByUic(int uic);

    List<ServicePoint> findAll();

    void saveAll(List<ServicePoint> servicePoints);

    void save(ServicePoint servicePoint);

}

File: backend/src/main/kotlin/ch/sbb/backend/servicepoints/domain/repository/ServicePointRepository.kt
package ch.sbb.backend.servicepoints.domain.repository

import ch.sbb.backend.servicepoints.domain.ServicePoint

interface ServicePointRepository {
    fun findByUic(uic: Int): ServicePoint?
    fun findAll(): List<ServicePoint>
    fun saveAll(servicePoints: List<ServicePoint>)
    fun save(servicePoint: ServicePoint)
}
